### DISCUSSION TITLE
Is using Path() params in a Depends() ok?

### DISCUSSION THREAD

**Discussion by jessekv** [fastapi/fastapi#6904] [Questions] [answered]
I want to validate a token based on what path is being accessed. I don't see anything similar in the documentation. Below is a simplified working example of the situation. My question is: Is this a valid usage, or is it abusing the fastapi API?

```py
from fastapi import Depends, FastAPI, Header, HTTPException, Path, status

app = FastAPI()

auth_table = {"me": "whee43", "you": "whaa32", "Hermoine": "SPEW4life"}

def user_validator(user: str = Path(...), x_token: str = Header(...)):
    if user in auth_table and auth_table[user] == x_token:
        return user
    raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED)

@app.get("/users/{user}")
async def get_user(valid_user=Depends(user_validator)):
    return {"user": valid_user}
```

**Comment by SnkSynthesis** [comment, marked as answer]
Yes, it is valid usage especially if you want to use `Path(...)` for validation, documentation, etc.

**Comment by jessekv** [comment]
Excellent, thanks!

**Comment by tiangolo** [comment]
Thanks for the help here @SnkSynthesis ! 👏  🙇 

And yep, that's fine! :heavy_check_mark: 

Thanks for reporting back and closing the issue @vdwees  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
